dev_tools/nyaml2nxdl/nyaml2nxdl_forward_tools.py

```diff
@@ -0,0 +1,1242 @@
+#!/usr/bin/env python3
+"""Creates an instantiated NXDL schema XML tree by walking the dictionary nest
+
+"""
+# -*- coding: utf-8 -*-
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+import textwrap
+import xml.etree.ElementTree as ET
+from xml.dom import minidom
+
+import yaml
+
+from ..utils import nexus as pynxtools_nxlib
+from .comment_collector import CommentCollector
+from .nyaml2nxdl_helper import LineLoader
+from .nyaml2nxdl_helper import cleaning_empty_lines
+from .nyaml2nxdl_helper import get_yaml_escape_char_reverter_dict
+from .nyaml2nxdl_helper import nx_name_type_resolving
+from .nyaml2nxdl_helper import remove_namespace_from_tag
+
+# pylint: disable=too-many-lines, global-statement, invalid-name
+DOM_COMMENT = (
+    "\n"
+    "# NeXus - Neutron and X-ray Common Data Format\n"
+    "# \n"
+    "# Copyright (C) 2014-2022 NeXus International Advisory Committee (NIAC)\n"
+    "# \n"
+    "# This library is free software; you can redistribute it and/or\n"
+    "# modify it under the terms of the GNU Lesser General Public\n"
+    "# License as published by the Free Software Foundation; either\n"
+    "# version 3 of the License, or (at your option) any later version.\n"
+    "#\n"
+    "# This library is distributed in the hope that it will be useful,\n"
+    "# but WITHOUT ANY WARRANTY; without even the implied warranty of\n"
+    "# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU\n"
+    "# Lesser General Public License for more details.\n"
+    "#\n"
+    "# You should have received a copy of the GNU Lesser General Public\n"
+    "# License along with this library; if not, write to the Free Software\n"
+    "# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA\n"
+    "#\n"
+    "# For further information, see http://www.nexusformat.org\n"
+)
+NX_CLSS = pynxtools_nxlib.get_nx_classes()
+NX_NEW_DEFINED_CLASSES = ["NX_COMPLEX"]
+NX_TYPE_KEYS = pynxtools_nxlib.get_nx_attribute_type()
+NX_ATTR_IDNT = "\\@"
+NX_UNIT_IDNT = "unit"
+DEPTH_SIZE = "    "
+NX_UNIT_TYPES = pynxtools_nxlib.get_nx_units()
+COMMENT_BLOCKS: CommentCollector
+CATEGORY = ""  # Definition would be either 'base' or 'application'
```

Comment:
I agree. The variable COMMENT_BLOCKS was utilised over the entire file without any modification after once it initialised. To prevent unnecessay copy over the function, it was globaly dified. I think fine in this aspect.